suites/suite_cam4.xml

```diff
@@ -0,0 +1,170 @@
+    <scheme>zm_diagnostics</scheme>
+
+    <!-- Check that energy and water change matches the boundary fluxes -->
+    <scheme>check_energy_scaling</scheme>
+    <scheme>check_energy_chng</scheme>
+
+    <!--
+      Hack shallow convection.
+      There are several nuances to the convect_shallow calling routine.
+      convect_shallow takes in total (from deep) convective cmfmc, qc, rliq and adds shallow convective tendencies
+      from Hack to them, also computing freqsh.
+      rliq is "reserved" liquid to be later added in macrophysics.
+
+      Tendencies are applied, then zm_conv_evap is ran to evaporate some of the precip directly into the environment to apply another set of tendencies.
+
+      This zm_conv_evap call takes as input some of the shallow convective quantities and adds to them, including
+      - precc -> prec (lwe_precipitation_rate_at_surface_due_to_..._convection)
+
+      Thus, an interstitial scheme to simply "convert" the shallow tendencies to the general tendencies in preparation for the zm_conv_evap call is also needed.
+      (hplin, 10/29/24)
+    -->
+
+    <!-- INITIALIZE DIAGNOSTICS (has init phase only) -->
+    <scheme>convect_shallow_diagnostics</scheme>
+
+    <!-- SHALLOW CONVECTION: HACK SCHEME -->
+    <scheme>check_energy_zero_fluxes</scheme>
+    <scheme>cmfmca</scheme>
+    <scheme>convect_shallow_diagnostics_after_shallow_scheme</scheme>
+    <scheme>apply_heating_rate</scheme>
+    <scheme>apply_constituent_tendencies</scheme>
+    <scheme>geopotential_temp</scheme>
+
+    <!-- SUBCLOUD EVAPORATION -->
+    <scheme>cloud_fraction_fice</scheme>
+    <!-- prepare state for zm_conv_evap (rename shallow outputs to generic inputs) -->
+    <!-- also zero out quantities going in/out -->
+    <scheme>set_shallow_conv_fluxes_to_general</scheme>
+    <scheme>zm_conv_evap</scheme>
+    <scheme>set_general_conv_fluxes_to_shallow</scheme>
+    <scheme>convect_shallow_diagnostics_after_convective_evaporation</scheme>
+    <scheme>apply_heating_rate</scheme>
+    <scheme>apply_constituent_tendencies</scheme>
+    <scheme>geopotential_temp</scheme>
+
+    <!-- DIAGNOSTIC QUANTITY UPDATES -->
+    <!-- eventually calculate shallow convective fraction diagnostic (FREQSH) -->
+    <!-- sum deep convective quantities with shallow including total convective rainout.
+         requires deep convective quantities to be actually available (i.e., rprddp) -->
+    <scheme>convect_shallow_sum_to_deep</scheme>
+    <scheme>convect_shallow_diagnostics_after_sum_to_deep</scheme>
+
+    <!-- Check that energy and water change matches the boundary fluxes -->
+    <scheme>check_energy_scaling</scheme>
+    <scheme>check_energy_chng</scheme>
+
+    <!--
+      RK Stratiform
+    -->
+
+    <!--
+      Radiation
+    -->
+
+  </group>
+
+  <group name="physics_after_coupler">
+
+    <!--
+      Vertical diffusion
+    -->
+
+    <!-- State diagnostics -->
+    <scheme>sima_state_diagnostics</scheme>
```

Comment:
I believe this is called before radiation in the `physics_before_coupler` group in CAM4.